### PR TITLE
Introduce a module with types/abstractions for working with VMSAs.

### PR DESCRIPTION
**Pull request by tylerfanelli**
Signed-off-by: Tyler Fanelli <tfanelli@redhat.com>

### CODE REVIEW

**Comment by crobinso** [comment]
LGTM, the additions too.

When making changes like this, it will be helpful in the future to have one commit that is just code movement from sevctl to sev, then additional commits on top that change things, so it's easier to find what changed.

**Review by crobinso — APPROVED** [review]

File: src/vmsa/mod.rs [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+    pub fn to_file(&self, filename: &str) -> VmsaRWResult {
+        let vmsa_buf = match bincode::serialize(self) {
+            Ok(v) => v,
+            Err(e) => return VmsaRWResult::serialization_err(e),
+        };
+
+        const SIZE: usize = std::mem::size_of::<Vmsa>();
+
+        // Pad to 4096 bytes
+        let buf: &mut [u8] = &mut [0; SIZE];
```
Comment (crobinso) [inline src/vmsa/mod.rs:482]: Just saw your sevctl PR and that the test suite is busted. SIZE here should be 4096, that will fix it

File: src/vmsa/mod.rs [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+
+        let vmsa = match bincode::deserialize(&data[..]) {
+            Ok(v) => v,
+            Err(e) => return VmsaRWResult::serialization_err(e),
+        };
+
+        VmsaRWResult::read_success(vmsa)
+    }
+
+    // Save binary VMSA content to the passed filename.
```
Comment (crobinso) [inline src/vmsa/mod.rs:471]: You can drop this comment, the one after it covers things

**Review by crobinso — CHANGES_REQUESTED** [review]

**Comment by tylerfanelli** [comment]
> LGTM, the additions too.
> 

Great, I'll merge.

> When making changes like this, it will be helpful in the future to have one commit that is just code movement from sevctl to sev, then additional commits on top that change things, so it's easier to find what changed.

Noted, I'll make sure to separate the transferring and additions in the future.

**Pull request merged**